ITE drivers/pwm: cleanup it8xxx2 pwm driver

RuibinChang opened the pull request:
1.Putting the PWM_CHANNEL_X, PWM_PRESCALER_CX information
  in the description.

2.Use the common definition EC_FREQ.

3.Use the common macro IT8XXX2_DT_ALT_ITEMS_LIST.

4.Stop using DRV_CONFIG, DRV_REG macro.

5.Delete pwmleds node in dts, because there isn't led in evb board.

Signed-off-by: Ruibin Chang <Ruibin.Chang@ite.com.tw>

RuibinChang commented:
As @brockus-zephyr  and @MaureenHelm suggestion in tachometer PR (https://github.com/zephyrproject-rtos/zephyr/pull/39993).

NOTE: This PR should  be merged after tachometer PR is merged.

keith-zephyr commented on boards/riscv/it8xxx2_evb/it8xxx2_evb.dts:
```diff
@@ -12,11 +12,6 @@
 	model = "IT8XXX2 EV-Board";
 	compatible = "riscv,it8xxx2-evb";
 
-	aliases {
```
Move the changes to the EVB board into a separate commit.

RuibinChang replied:
Done

MaureenHelm reviewed (changes requested):
Please rebase to pick up a fix for the buildkite failure, Also, please address the compliance failure with your commit message.

RuibinChang commented on drivers/pwm/pwm_ite_it8xxx2.c:
```diff
@@ -71,15 +69,16 @@ static void pwm_enable(const struct device *dev, int enabled)
-	struct pwm_it8xxx2_regs *const inst = DRV_REG(dev);
+	const struct pwm_it8xxx2_cfg *config = dev->config;
+	struct pwm_it8xxx2_regs *const inst =
+				(struct pwm_it8xxx2_regs *) config->base;
```
Done.

I had re-signed the two commit by "sign off" button of git gui,  and I check that sign and e-mail address of both commit are the same. I'm finding where the error comes from.

RuibinChang replied:
Commit message failure is done.

gmarull commented on dts/bindings/pwm/ite,it8xxx2-pwm.yaml:
```diff
@@ -21,14 +21,17 @@ properties:
+            - 5
+            - 6
+            - 7
+      description: 0 = PWM_CHANNEL_0, 1 = PWM_CHANNEL_1, 2 = PWM_CHANNEL_2,
```
use `description: |` for multiline descriptions

RuibinChang replied:
Ack. Done.

RuibinChang committed:
ITE drivers/pwm: cleanup it8xxx2 pwm driver

1.Putting the PWM_CHANNEL_X, PWM_PRESCALER_CX information
  in the description.
2.Use the common definition EC_FREQ.
3.Use the common macro IT8XXX2_DT_ALT_ITEMS_LIST.
4.Stop using DRV_CONFIG, DRV_REG macro.

Signed-off-by: Ruibin Chang <Ruibin.Chang@ite.com.tw>

RuibinChang committed:
ITE borads/riscv/it8xxx2_evb: delete pwmleds node in dts

Delete pwmleds node in dts, because there isn't led on evb board.

Signed-off-by: Ruibin Chang <Ruibin.Chang@ite.com.tw>

jackrosenthal commented on drivers/pwm/pwm_ite_it8xxx2.c:
```diff
@@ -13,16 +13,26 @@
 
 #define PWM_CTRX_MIN	100
-#define PWM_EC_FREQ	MHZ(8)
+#define PWM_FREQ	EC_FREQ
```
I get compile errors when compiling this code:

```
ERROR: [/mnt/host/source/src/platform/ec/build/zephyr/hayato:ro]/home/jrosenth/zephyr/drivers/pwm/pwm_ite_it8xxx2.c:23:25: error: 'EC_FREQ' undeclared (first use in this function); did you mean 'ECREG'?
ERROR: [/mnt/host/source/src/platform/ec/build/zephyr/hayato:ro]   23 | #define PWM_FREQ        EC_FREQ
ERROR: [/mnt/host/source/src/platform/ec/build/zephyr/hayato:ro]      |                         ^~~~~~~
ERROR: [/mnt/host/source/src/platform/ec/build/zephyr/hayato:ro]/home/jrosenth/zephyr/drivers/pwm/pwm_ite_it8xxx2.c:80:38: note: in expansion of macro 'PWM_FREQ'
ERROR: [/mnt/host/source/src/platform/ec/build/zephyr/hayato:ro]   80 |                 *cycles = (uint64_t) PWM_FREQ;
ERROR: [/mnt/host/source/src/platform/ec/build/zephyr/hayato:ro]      |                                      ^~~~~~~~
ERROR: [/mnt/host/source/src/platform/ec/build/zephyr/hayato:ro]/home/jrosenth/zephyr/drivers/pwm/pwm_ite_it8xxx2.c:23:25: note: each undeclared identifier is reported only once for each function it appears in
ERROR: [/mnt/host/source/src/platform/ec/build/zephyr/hayato:ro]   23 | #define PWM_FREQ        EC_FREQ
ERROR: [/mnt/host/source/src/platform/ec/build/zephyr/hayato:ro]      |                         ^~~~~~~
ERROR: [/mnt/host/source/src/platform/ec/build/zephyr/hayato:ro]/home/jrosenth/zephyr/drivers/pwm/pwm_ite_it8xxx2.c:80:38: note: in expansion of macro 'PWM_FREQ'
ERROR: [/mnt/host/source/src/platform/ec/build/zephyr/hayato:ro]   80 |                 *cycles = (uint64_t) PWM_FREQ;
ERROR: [/mnt/host/source/src/platform/ec/build/zephyr/hayato:ro]      |                                      ^~~~~~~~
```

I don't see any references to `EC_FREQ` in the tree. What am I missing?

RuibinChang replied:
The EC_FREQ is defined in tachometer PR (https://github.com/zephyrproject-rtos/zephyr/pull/39993), I think that tachometer PR is ready to merge.

MaureenHelm replied:
This suggests that this driver is not getting built in CI. @RuibinChang will you please look into enabling `tests/drivers/pwm/` for an ITE board?

RuibinChang replied:
I build board it8xxx2_evb with pwm driver, it doesn't show the error.
